Userland/Utilities/test-imap.cpp

```diff
@@ -38,9 +37,11 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     if (interactive_password) {
         password = TRY(Core::get_password());
     } else {
-        auto standard_input = Core::DeprecatedFile::standard_input();
-        password = Core::SecretString::take_ownership(standard_input->read_all());
+        auto standard_input = TRY(Core::File::standard_input());
+        // This might leave the clear password in unused memory, but this is only a test program anyway.
+        password = Core::SecretString::take_ownership(TRY(standard_input->read_until_eof()));
     }
+    password = TRY(Core::get_password());
```

Comment:
line 44 looks out of place... won't this get the password twice?

It's also unclear if anyone was intending to script this file, and whether a non-interactive mode is even desired 🤷 .

I also don't know how to test this program, so after this one I think we can go ahead and merge it to remove the last user of Core::DeprecatedFile::standard_input()...

Reply:
You're right, I forgot this line while testing the I/O behavior of this block.

The password is the last thing that this program reads from stdin, and there's the `interactive_password` flag. I think this makes it clear that a scripted, non-interactive mode was intentional.

I just don't know how to actually run the program against a server; trying it out with my real-life e-mail addresses is a no-go, since I don't want it to mess with my e-mail state. That's why I'm still hoping that either of the original authors respond. If they don't, I guess we could make another PR deleting the test program entirely, but that's a thing for the future. :shrug: